nemo/collections/tts/models/base.py

```diff
@@ -96,7 +100,7 @@ def list_available_models(cls) -> 'List[PretrainedModelInfo]':
 
 class GlowVocoder(Vocoder):
     """ Base class for all Vocoders that use a Glow or reversible Flow-based setup. All child class are expected
-    to have a parameter called audio_to_melspec_precessor that is an instance of
+    to have a parameter called audio_to_melspec_processor that is an instance of
```

Comment:
Oh boy, I remember this. The parameter _is_ called `audio_to_melspec_precessor` (sic) and we debated correcting it to `preprocessor` a while ago, but ended up deciding not to since it'd break a bunch of ASR and TTS models.

I'd say to leave the typo as-is here and we can revisit the discussion with the ASR team involved. It does still bother me a bit every time I see it :P

Reply:
yeah, I updated it with a thorough check, and fixed all. It should work for all now.

Reply:
I don't think ASR still used this func anymore.

Reply:
Can you double-check that loading older models works with this? I think that was the sticking point in the past. If backwards-compatibility works fine then it should be all okay!

Reply:
Good point. Will do.

Reply:
you are right, our CI captured the error when changing names,
```shell-session
______ ERROR at setup of TestExportable.test_HifiGanModel_export_to_onnx _______



    @pytest.fixture()

    def hifigan_model():

>       model = HifiGanModel.from_pretrained(model_name="tts_hifigan")



tests/collections/tts/test_tts_exportables.py:30: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

nemo/core/classes/common.py:850: in from_pretrained

    instance = class_.restore_from(

nemo/core/classes/modelPT.py:316: in restore_from

    instance = cls._save_restore_connector.restore_from(

nemo/core/connectors/save_restore_connector.py:242: in restore_from

    self.load_instance_with_state_dict(instance, state_dict, strict)

nemo/core/connectors/save_restore_connector.py:197: in load_instance_with_state_dict

    instance.load_state_dict(state_dict, strict=strict)

nemo/collections/tts/models/hifigan.py:421: in load_state_dict

    super().load_state_dict(new_state_dict, strict=strict)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 



self = HifiGanModel(

  (audio_to_melspec_preprocessor): FilterbankFeatures()

  (trg_melspec_fn): FilterbankFeatures()

  (gene...(feature_loss): FeatureMatchingLoss()

  (discriminator_loss): DiscriminatorLoss()

  (generator_loss): GeneratorLoss()

)

state_dict = OrderedDict([('audio_to_melspec_precessor.window', tensor([0.0000e+00, 9.4175e-06, 3.7730e-05,  ..., 3.7730e-05, 9.417...6,  0.0592,  0.2073],

         [-0.1876, -0.0347, -0.1377],

         [-0.2356, -0.1337, -0.1584]]], device='cuda:0'))])

strict = True



    def load_state_dict(self, state_dict: Mapping[str, Any],

                        strict: bool = True):

        r"""Copies parameters and buffers from :attr:`state_dict` into

        this module and its descendants. If :attr:`strict` is ``True``, then

        the keys of :attr:`state_dict` must exactly match the keys returned

        by this module's :meth:`~torch.nn.Module.state_dict` function.

    

        Args:

            state_dict (dict): a dict containing parameters and

                persistent buffers.

            strict (bool, optional): whether to strictly enforce that the keys

                in :attr:`state_dict` match the keys returned by this module's

                :meth:`~torch.nn.Module.state_dict` function. Default: ``True``

    

        Returns:

            ``NamedTuple`` with ``missing_keys`` and ``unexpected_keys`` fields:

                * **missing_keys** is a list of str containing the missing keys

                * **unexpected_keys** is a list of str containing the unexpected keys

    

        Note:

            If a parameter or buffer is registered as ``None`` and its corresponding key

            exists in :attr:`state_dict`, :meth:`load_state_dict` will raise a

            ``RuntimeError``.

        """

        if not isinstance(state_dict, Mapping):

            raise TypeError("Expected state_dict to be dict-like, got {}.".format(type(state_dict)))

    

        missing_keys: List[str] = []

        unexpected_keys: List[str] = []

        error_msgs: List[str] = []

    

        # copy state_dict so _load_from_state_dict can modify it

        metadata = getattr(state_dict, '_metadata', None)

        state_dict = OrderedDict(state_dict)

        if metadata is not None:

            # mypy isn't aware that "_metadata" exists in state_dict

            state_dict._metadata = metadata  # type: ignore[attr-defined]

    

        def load(module, prefix=''):

            local_metadata = {} if metadata is None else metadata.get(prefix[:-1], {})

            module._load_from_state_dict(

                state_dict, prefix, local_metadata, True, missing_keys, unexpected_keys, error_msgs)

            for name, child in module._modules.items():

                if child is not None:

                    load(child, prefix + name + '.')

    

            # Note that the hook can modify missing_keys and unexpected_keys.

            incompatible_keys = _IncompatibleKeys(missing_keys, unexpected_keys)

            for hook in module._load_state_dict_post_hooks.values():

                out = hook(module, incompatible_keys)

                assert out is None, (

                    "Hooks registered with ``register_load_state_dict_post_hook`` are not"

                    "expected to return new values, if incompatible_keys need to be modified,"

                    "it should be done inplace."

                )

    

        load(self)

        del load

    

        if strict:

            if len(unexpected_keys) > 0:

                error_msgs.insert(

                    0, 'Unexpected key(s) in state_dict: {}. '.format(

                        ', '.join('"{}"'.format(k) for k in unexpected_keys)))

            if len(missing_keys) > 0:

                error_msgs.insert(

                    0, 'Missing key(s) in state_dict: {}. '.format(

                        ', '.join('"{}"'.format(k) for k in missing_keys)))

    

        if len(error_msgs) > 0:

>           raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(

                               self.__class__.__name__, "\n\t".join(error_msgs)))

E           RuntimeError: Error(s) in loading state_dict for HifiGanModel:

E           	Missing key(s) in state_dict: "audio_to_melspec_preprocessor.window", "audio_to_melspec_preprocessor.fb". 

E           	Unexpected key(s) in state_dict: "audio_to_melspec_precessor.window", "audio_to_melspec_precessor.fb".
```

Reply:
I will decouple this change into another PR later.